### PR TITLE
Add table for projections

### PR DESCRIPTION
**Pull request by willschlitzer**
As discussed in #746, this is a table listing the projections and the associated arguments for them.

### CODE REVIEW

**Comment by willschlitzer** [comment]
@seisman I'm a bit stuck. I tried combining the title format from the other README docs with the table formatting in the GMT table code. Any idea why my code is causing trouble?

**Comment by seisman** [comment]
The error message is:
```
If you use an explicit folder ordering, you must specify all folders. Explicit order not found for ..\examples\projections\table
```

It seems you need to add `..\examples\projections\table` to:

https://github.com/GenericMappingTools/pygmt/blob/600d9aec47105502745b88772d2ef3f923981687/doc/conf.py#L61-L70

File: examples/projections/table/README.txt [hunk 1]
```diff
@@ -0,0 +1,84 @@
+Projection Table
+----------------
+
+PyGMT offers 31 map projections. These are specified using the ``projection`` argument.
+The projection codes for the projection styles are shown below. Upper case arguments
+set *map width* and lower case arguments set *map scale*.
```
Comment (seisman) [inline examples/projections/table/README.txt:6]: > Upper case arguments set *map width* and lower case arguments set *map scale*.

New users will be confused about what "Upper case" and "Lower case" mean. Please try to rephrase it, or just give an example.

Comment (seisman) [reply examples/projections/table/README.txt:6]: https://pygmt-git-fork-willschlitzer-proj-table.gmt.vercel.app/projections/index.html

I just realized that we already have similar sentences at the top of this page.

> PyGMT support many map projections. Use the projection argument to specify which one you want to use in all plotting modules. The projection is specified by a one letter code along with (sometimes optional) reference longitude and latitude and the width of the map (for example, Alon0/lat0[/horizon]/width). The map height is determined based on the region and projection.

Perhaps no need to repeat it in the `projections/table/README.txt` file

Comment (willschlitzer) [reply examples/projections/table/README.txt]: I removed it from the table README

File: examples/projections/table/README.txt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+     - General perspective
+   * - **H**\ [|lon0|/]\ *width*
+     - Hammer equal area
+   * - **I**\ [|lon0|/]\ *width*
+     - Sinusoidal equal area
+   * - **J**\ [|lon0|/]\ *width*
+     - Miller cylindrical
+   * - **Kf**\ [|lon0|/]\ *width*
+     - Eckert IV equal area
+   * - **-Ks**\ [|lon0|/]\ *width*
```
Comment (seisman) [inline examples/projections/table/README.txt:48]:
```suggestion
   * - **Ks**\ [|lon0|/]\ *width*
```

File: examples/projections/table/README.txt [hunk 3]
```diff
@@ -0,0 +1,83 @@
+Projection Table
+----------------
+
+PyGMT offers 31 map projections. These are specified using the ``projection`` argument.
+The projection codes for the projection styles are shown below.
```
Comment (seisman) [inline examples/projections/table/README.txt:5]: There are similar sentences at the top of this page (https://pygmt-f6khtwuqs.vercel.app/projections/index.html).
```suggestion
The below table shows the projection codes for the 31 GMT projections.
```

Comment (seisman) [reply examples/projections/table/README.txt:5]: @willschlitzer I'm not a native English speaker, you can rewrite the sentence if it doesn't read well.

Comment (willschlitzer) [reply examples/projections/table/README.txt]: Looks good to me!

**Pull request merged**